Upgrade guide for KFP standalone

Bobgy opened the pull request:
Preview of updated content: https://deploy-preview-1177--competent-brattain-de2d6d.netlify.com/docs/pipelines/pipelines-deploy-alternatives/#best-practice-maintaining-custom-manifests

/assign @IronPan 
/cc @jessiezcc

This is part of https://github.com/kubeflow/pipelines/issues/1638

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/website/1177)
<!-- Reviewable:end -->

sarahmaddox commented:
/assign

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+
+# You can add other customizations here using kustomize.
+
+# Edit ref in the following link to deploy a different version of KFP.
```
It's not clear what "KFP" means, to folks outside the development team. Change "KFP" to "Kubeflow Pipelines", here and below.

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
````diff
@@ -0,0 +1,178 @@
+- github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref={{% kfp-latest-version %}}
+```
+
+### How to?
````
It's not clear what this heading means. Please update the wording to better describe the purpose of this section, and remove the question mark.

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+### How to?
+* Deploy: `kubectl apply -k $YOUR_REPO`
+* Upgrade
+    1. (Recommended) backup your data storages for KFP.
```
s/backup/Back up

(Two words; start with a capital letter.)

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+
+### How to?
+* Deploy: `kubectl apply -k $YOUR_REPO`
+* Upgrade
```
For improved clarity, add a colon ":" at the end of this line and all similar lines below.

Also, please add words indicating what people are upgrading.

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+    Check [Kubeflow Pipelines github repo](https://github.com/kubeflow/pipelines/releases) for available releases.
+    1. Deploy: `kubectl apply -k $YOUR_REPO`
+* Downgrade
+    1. Make sure you made a backup before upgrading.
```
s/made/make

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+    1. Deploy: `kubectl apply -k $YOUR_REPO`
+* Downgrade
+    1. Make sure you made a backup before upgrading.
+    1. Delete the deployment `kubectl delete -k $YOUR_REPO`.
```
s/deployment/deployment:

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+
+
+### Further reading
+kustomize's [recommended workflow using an Off-The-Shelf configuration](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/workflows.md#off-the-shelf-configuration)
```
Remove capital letters.

sarahmaddox commented on content/docs/pipelines/pipelines-deploy-alternatives.md:
```diff
@@ -0,0 +1,178 @@
+
+### Further reading
+kustomize's [recommended workflow using an Off-The-Shelf configuration](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/workflows.md#off-the-shelf-configuration)
+like KFP lite we provide.
```
It's not clear what this means: "like KFP lite we provide." Please clarify the wording.

sarahmaddox commented on content/docs/pipelines/upgrade.md:
```diff
@@ -11,8 +11,13 @@ supports the following capabilities:
+Note that upgrade isn't currently supported for Kubeflow deployment, check
+[this issue](https://github.com/kubeflow/kubeflow/issues/3727) for progress.
+
+We provide an alternative lite deployment to deploy Kubeflow Pipelines. It is
```
s/lite/lightweight
(Unless branding decision says otherwise. See my comments in related PR.)

sarahmaddox commented on content/docs/pipelines/upgrade.md:
```diff
@@ -11,8 +11,13 @@ supports the following capabilities:
+[this issue](https://github.com/kubeflow/kubeflow/issues/3727) for progress.
+
+We provide an alternative lite deployment to deploy Kubeflow Pipelines. It is
+documented [here](docs/pipelines/pipelines-deploy-alternatives). This lite
```
Link is broken.

Also, instead of "here" as link text, provide a description of the doc you're linking to.

Bobgy commented:
Made a quick update to merge with master.
/hold

Bobgy commented:
@IronPan can you help me review? Especially downgrade part.

IronPan commented on content/docs/pipelines/standalone-deployment-gcp.md:
```diff
@@ -152,6 +152,40 @@ kubectl delete -k manifests/kustomize/env/dev
+# You can add other customizations here using kustomize.
+# Edit ref in the following link to deploy a different version of Kubeflow Pipelines.
+bases:
+- github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref={{% kfp-latest-version %}}
```
// -> /

Bobgy replied:
I understand it looks supicious, but this is intended url format, reference: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md#url-format

IronPan commented on content/docs/pipelines/standalone-deployment-gcp.md:
```diff
@@ -152,6 +152,40 @@ kubectl delete -k manifests/kustomize/env/dev
+* Deploy: `kubectl apply -k $YOUR_REPO`
+* Upgrade:
+    1. (Recommended) backup your data storages for KFP.
+    1. Edit `ref={{% kfp-latest-version %}}` to a version you want to upgrade to.
```
Is it possible to use "kustomize edit set" instead for changing the versioning?

Bobgy replied:
Not sure, I haven't tried that.

I preferred manually editing it, because we can skip asking users to download `kustomize` tool.

IronPan commented on content/docs/pipelines/standalone-deployment-gcp.md:
```diff
@@ -152,6 +152,40 @@ kubectl delete -k manifests/kustomize/env/dev
+    1. Edit `ref={{% kfp-latest-version %}}` to a version you want to upgrade to.
+￼    Check [Kubeflow Pipelines github repo](https://github.com/kubeflow/pipelines/releases) for available releases.
+￼    1. Deploy: `kubectl apply -k $YOUR_REPO`.
+￼* Downgrade:
```
I don't think we support downgrade as it can causes uncertain behavior. e.g. new version api that @jingzhang36 is working on.

Bobgy replied:
Note I mentioned making a backup and first restoring PVs to backed up state before downgrade.
I think this is doable, but I need to actually try it and prepare detailed instructions.

I now change my mind and prefer skipping downgrade instructions for now.
What do you think?

IronPan replied:
sg

Bobgy commented:
@IronPan thanks for the review!

sarahmaddox commented on content/docs/pipelines/standalone-deployment-gcp.md:
````diff
@@ -84,6 +84,15 @@ document better and be able to customize based on your needs.
     ```
 
+## Upgrade
+1. Make sure kubectl talks to your cluster.
````
We should tell them how to do this. (Even though we tell them higher up on the page, people may come directly to this section, and then not know what to do.) I think you can copy lines 69-72.

Bobgy commented:
UPDATE, new predeployed link: https://deploy-preview-1177--competent-brattain-de2d6d.netlify.com/docs/pipelines/standalone-deployment-gcp/#upgrade

@sarahmaddox do you want to do another round of review?

sarahmaddox commented on content/docs/pipelines/standalone-deployment-gcp.md:
````diff
@@ -152,6 +161,36 @@ kubectl delete -k manifests/kustomize/env/dev
+- github.com/kubeflow/pipelines//manifests/kustomize/env/dev?ref={{% kfp-latest-version %}}
+```
+
+### How to
````
Should this entire section disappear now that we have an upgrade section above? If we need to keep this section, then we should explain how it differs from the earlier upgrade section - is it for upgrades when using a custom manifest?

sarahmaddox replied:
Looks good!

sarahmaddox commented on content/docs/pipelines/standalone-deployment-gcp.md:
```diff
@@ -152,6 +161,36 @@ kubectl delete -k manifests/kustomize/env/dev
+* Upgrade:
+    1. (Recommended) backup your data storages for KFP.
+    1. Edit `ref={{% kfp-latest-version %}}` to a version you want to upgrade to.
+￼    Check [Kubeflow Pipelines github repo](https://github.com/kubeflow/pipelines/releases) for available releases.
```
There are some rendering issues on the page in this section. See the preview:
https://deploy-preview-1177--competent-brattain-de2d6d.netlify.com/docs/pipelines/standalone-deployment-gcp/#how-to

sarahmaddox commented:
Preview is now here: https://deploy-preview-1177--competent-brattain-de2d6d.netlify.com/docs/pipelines/standalone-deployment-gcp/

Bobgy commented:
@sarahmaddox Thanks for the review! I updated according to your comments. Please let me know if it looks good now or not.

Note that I also added a pointer in Pipelines upgrade doc to point to standalone deployment, because that's the only deployment that have upgrade support from pipelines team.

Bobgy commented:
/hold cancel

sarahmaddox commented on content/docs/pipelines/standalone-deployment-gcp.md:
```diff
@@ -152,6 +161,36 @@ kubectl delete -k manifests/kustomize/env/dev
+* Deploy: `kubectl apply -k $YOUR_REPO`
+* Upgrade:
+    1. (Recommended) backup your data storages for KFP.
+    1. Edit `ref={{% kfp-latest-version %}}` to a version you want to upgrade to.
```
There's still something odd happening in the rendering of the page at this point. It has a box with the word "Obj" on it:
https://deploy-preview-1177--competent-brattain-de2d6d.netlify.com/docs/pipelines/standalone-deployment-gcp/#how-to-deploy-upgrade-and-uninstall-using-the-repo

Bobgy replied:
Thanks for the catch! I've fixed it.

sarahmaddox commented:
Thanks!

/lgtm
/approve
